### PR TITLE
runtimevar/etcdvar: implement runtimevar for etcd

### PR DESCRIPTION
**Pull request by vangent**
@zombiezen This PR adds a bunch of dependencies for etcd. Is this OK?

### CODE REVIEW

**Comment by vangent** [comment]
FYI, the Travis build is failing because of the new dependencies.

**Comment by zombiezen** [comment]
You'll need to add these dependencies to the GCS bucket after verifying that they are each covered under one of the `notice`, `permissive`, or `unencumbered` licences under the categories defined by [Google Open Source](https://opensource.google.com/docs/thirdparty/licenses/).

File: runtimevar/etcdvar/etcdvar.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+			// New Value
+			val, err := i.decoder.Decode(kv.Value)
+			if err != nil {
+				return checkSameErr(err)
+			}
+			return &driver.Variable{Value: val, UpdateTime: time.Now()}, kv.Version, 0, nil
+		}
+
+		// Value hasn't changed. Wait for change events.
+		_ = <-ch
```
Comment (shantuo) [inline runtimevar/etcdvar/etcdvar.go]: Just `<-ch`? Or is it a lint check thing?

File: go.mod [hunk 2]
```diff
@@ -21,34 +21,60 @@ require (
 	github.com/aws/aws-sdk-go v1.13.20
 	github.com/aws/aws-xray-sdk-go v1.0.0-rc.5 // indirect
 	github.com/basvanbeek/ocsql v0.0.0-20180627133915-d946a62502e3
+	github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973 // indirect
 	github.com/census-ecosystem/opencensus-go-exporter-aws v0.0.0-20180411051634-41633bc1ff6b
+	github.com/coreos/bbolt v1.3.1-coreos.6 // indirect
+	github.com/coreos/etcd v3.3.9+incompatible
```
Comment (shantuo) [inline go.mod]: Is it the same thing as `go.etcd.io/etcd`? Just curious, but this seems to be a transitive dependency though.

**Comment by vangent** [comment]
Fixes #267

File: runtimevar/etcdvar/etcdvar.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+func New(name string, cli *clientv3.Client, decoder *runtimevar.Decoder) (*runtimevar.Variable, error) {
+	return runtimevar.New(&impl{
+		name:    name,
+		client:  cli,
+		decoder: decoder,
+	}), nil
+}
+
+// impl implements driver.Watcher.
+type impl struct {
```
[the remark this answers is not in the document]
Comment (vangent) [reply runtimevar/etcdvar/etcdvar.go]: Done. I left the comment, I think it's good to be explicit.

File: runtimevar/etcdvar/etcdvar.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+	name    string
+	client  *clientv3.Client
+	decoder *runtimevar.Decoder
+}
+
+func (i *impl) WatchVariable(ctx context.Context, prevVersion interface{}, prevErr error) (*driver.Variable, interface{}, time.Duration, error) {
+
+	// checkSameErr checks to see if err is the same as prevErr, andif so, returns
```
[the remark this answers is not in the document]
Comment (vangent) [reply runtimevar/etcdvar/etcdvar.go]: DOne.

File: runtimevar/etcdvar/etcdvar.go [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+	// the "no change" signal with w.waitTime.
+	checkSameErr := func(err error) (*driver.Variable, interface{}, time.Duration, error) {
+		if prevErr != nil {
+			if err.Error() == prevErr.Error() {
```
[the remark this answers is not in the document]
Comment (vangent) [reply runtimevar/etcdvar/etcdvar.go]: Done.

File: runtimevar/etcdvar/etcdvar.go (hunk 1 above)
Comment (vangent) [reply runtimevar/etcdvar/etcdvar.go]: Done. We'll see if the linter complains.

File: go.mod (hunk 2 above)
Comment (vangent) [reply go.mod]: It looks like they are moving everything to go.etcd.io/etcd, but not everything has had a release with that updated yet, so we're pulling in both. I updated to use the old path consistently for now.

File: go.mod [hunk 6]
```diff
@@ -21,34 +21,59 @@ require (
...
```
Comment (vangent) [inline go.mod]: I've confirmed all the licenses are OK (they are all "notice"):

```
github.com/beorn7/perks: MIT ("notice")
github.com/coreos/bbolt: MIT ("notice")
github.com/coreos/etcd: Apache 2.0 ("notice")
github.com/coreos/go-semver: Apache 2.0 ("notice")
github.com/coreos/go-systemd: Apache 2.0 ("notice")
github.com/coreos/pkg: Apache 2.0 ("notice")
github.com/dgrijalva/jwt-go: MIT ("notice")
github.com/ghodss/yaml: MIT ("notice")
github.com/gogo/protobuf: Go Authors ("notice")
github.com/google/btree: Apache 2.0 ("notice")
github.com/gorilla/websocket: BSD 2-Clause ("notice")
github.com/grpc-ecosystem/go-grpc-prometheus: Apache 2.0 ("notice")
github.com/grpc-ecosystem/grpc-gateway: BSD 3-Clause ("notice")
github.com/jonboulle/clockwork: Apache 2.0 ("notice")
github.com/matttproud/golang_protobuf_extensions: Apache 2.0 ("notice")
github.com/prometheus/client_golang: Apache 2.0 ("notice")
github.com/prometheus/client_model: Apache 2.0 ("notice")
github.com/prometheus/common: Apache 2.0 ("notice")
github.com/prometheus/procfs: Apache 2.0 ("notice")
github.com/sirupsen/logrus: MIT ("notice")
github.com/soheilhy/cmux: Apache 2.0 ("notice")
github.com/tmc/grpc-websocket-proxy: MIT ("notice")
github.com/ugorji/go/codec: MIT ("notice")
github.com/xiang90/probing: MIT ("notice")
golang.org/x/time: Go Authors ("notice")
```

**Pull request merged**